### ISSUE TITLE
[simage] build failure

### ISSUE THREAD

**Issue by ChrisSchulte** [microsoft/vcpkg#23915]
**Host Environment**
- OS: Windows 10
- Compiler: Visual Studio 2019

**To Reproduce**
`./vcpkg install simage`

**Failure logs**
```
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j9
    Working Directory: D:/cschulte/DevEnv64/vcpkg/buildtrees/simage/x64-windows-dbg
    See logs for more information:
      D:\cschulte\DevEnv64\vcpkg\buildtrees\simage\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/simage/portfile.cmake:62 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)


Error: Building package simage:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+simage
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[simage]+Build+error
including:
  package: simage[all,avienc,core,gdiplus,oggvorbis,sndfile]:x64-windows -> 1.8.1#4
    vcpkg-tool version: 2022-03-25-d9702d9d89b9468c32efc3f18fb84c3021d7dbae
    vcpkg-scripts version: 9d9a6f486 2022-03-30 (31 hours ago)
```

As can be seen in the config log GdiPlus and Vfw refer to Windows Kits 8.1 knowing that Windows Kits 10 is also installed. After digging, and comparing behaviour to the one described in issue #22941 the problem can be solved by reverting the HINTS order in FindGdiPlus.cmake and FindVfw.cmake so that the most recent Windows Kits comes first in the search order instead of the oldest one first as it is now.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/8395596/install-x64-windows-dbg-out.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/8395602/config-x64-windows-out.log)

**Comment by LilyWangLL** [comment]
https://github.com/microsoft/vcpkg/issues/22941#issuecomment-1085529675

**Comment by JackBoosY** [comment]
It seems is a `FindVfw.cmake` /  `FindGdiPlus.cmake` bug.

**Comment by ChrisSchulte** [comment]
Both are provided by the simage project itself.

**Comment by ChrisSchulte** [comment]
@JackBoosY, I've cloned the simage repo directly to make some tests and changing the order of HINTS doesn't change anything because as the PATH_SUFFIXES in both cmake files don't match the path construction of SDK 10. Indeed for SDK 8.1 the path will be 8.1/Include/um but for the SDK 10 it will be 10/Include/SDKVersion/um with multiple choices for the SDK version... Working from a x64 Native Tools Command Prompt without any hints nor path_suffixes in both cmake files makes the code compile flawlessly. Working from a standard command prompt it will not find neither GdiPlus nor Vfw.
I think I will open an issue on simage side to see what would be their ideas and workarounds.

**Comment by ChrisSchulte** [comment]
issue opened on simage side [https://github.com/coin3d/simage/issues/55#issue-1189744105](https://github.com/coin3d/simage/issues/55#issue-1189744105)

**Comment by ChrisSchulte** [comment]
For those who want to be able to workaround temporary, add following two patches to simage portfile.cmake 

[simage-patches.zip](https://github.com/microsoft/vcpkg/files/8397971/simage-patches.zip)

**Comment by dg0yt** [comment]
The easiest fix for the port would be a `CMAKE_PROJECT_INCLUDE` script, and doing a `find_path`/`find_library`/`select_library_configuration` there. Plain cmake to set the input variables, no patch.

**Comment by JackBoosY** [comment]
> The issue has been resolved on simage side with release 1.8.2 four days ago. Updating port file should correct the behaviour.

Can you please provide the upstream PR? Thanks.
